src/liballoc/boxed.rs

```diff
@@ -125,6 +153,111 @@ impl<T> Box<T> {
     }
 }
 
+impl<T> Box<[T]> {
+    /// Construct a new boxed slice with uninitialized contents.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(new_uninit)]
+    ///
+    /// let mut values = Box::<[u32]>::new_uninit_slice(3);
+    ///
+    /// let values = unsafe {
+    ///     // Deferred initialization:
+    ///     values[0].as_mut_ptr().write(1);
+    ///     values[1].as_mut_ptr().write(2);
+    ///     values[2].as_mut_ptr().write(3);
+    ///
+    ///     values.assume_init()
+    /// };
+    ///
+    /// assert_eq!(*values, [1, 2, 3])
+    /// ```
+    #[unstable(feature = "new_uninit", issue = "0")]
+    pub fn new_uninit_slice(len: usize) -> Box<[mem::MaybeUninit<T>]> {
+        let layout = alloc::Layout::array::<mem::MaybeUninit<T>>(len).unwrap();
+        let ptr = unsafe { alloc::alloc(layout) };
```

Comment:
Wouldn't it make more sense to call `exchange_malloc`, just to make sure this is allocated the same way as any other box?

Reply:
I’ve changed this to use `Global`, to rely on this documented guarantee: https://doc.rust-lang.org/beta/std/boxed/index.html#memory-layout

Reply:
But why wouldn't you just re-use `exchange_malloc`?

Reply:
It’s less convenient to use with `Layout::array`, and everything else in this module is already using `Global`.

Reply:
All right. Together with the documented guarantee, that seems fair.